crates/server/src/admin/handlers/context/invite_to_context.rs

```diff
@@ -3,14 +3,33 @@ use std::sync::Arc;
 use axum::response::IntoResponse;
 use axum::{Extension, Json};
 use calimero_server_primitives::admin::{InviteToContextRequest, InviteToContextResponse};
+use reqwest::StatusCode;
 
-use crate::admin::service::{parse_api_error, ApiResponse};
+use crate::admin::service::{parse_api_error, ApiError, ApiResponse};
 use crate::AdminState;
 
 pub async fn handler(
     Extension(state): Extension<Arc<AdminState>>,
     Json(req): Json<InviteToContextRequest>,
 ) -> impl IntoResponse {
+    let has_permission = state
+        .ctx_manager
+        .has_invite_permission(req.context_id, req.inviter_id)
+        .await
+        .map_err(parse_api_error);
+
+    match has_permission {
+        Ok(false) => {
+            return ApiError {
+                status_code: StatusCode::FORBIDDEN,
+                message: "User does not have permission to invite".to_owned(),
+            }
+            .into_response()
+        }
+        Err(err) => return err.into_response(),
+        _ => (),
+    }
```

Comment:
1. permissions checks are managed by the context config contract
2. the server is just one way to interact with the node, of which ALL ways must be behaviorally consistent, so we don't have to maintain the same code across multiple fronts and employ a manual effort of keeping them synchronized. which means, no functional logic beyond calling to the node should be done in server handlers. this will also let you reuse the invitation logic from `merod` for example.